src/main/java/com/agnieszkapawska/flashcards/domain/facades/FlashcardFacade.java

```diff
@@ -4,24 +4,32 @@
 import com.agnieszkapawska.flashcards.domain.dtos.FlashcardSaveResponseDto;
 import com.agnieszkapawska.flashcards.domain.exceptions.EntityNotCreatedException;
 import com.agnieszkapawska.flashcards.domain.models.Flashcard;
-import com.agnieszkapawska.flashcards.domain.repositories.FlashcardRepository;
+import com.agnieszkapawska.flashcards.domain.models.QuestionTag;
+import com.agnieszkapawska.flashcards.domain.services.FlashcardService;
+import com.agnieszkapawska.flashcards.domain.services.QuestionTagService;
 import lombok.AllArgsConstructor;
 import org.modelmapper.ModelMapper;
 import org.springframework.dao.DataIntegrityViolationException;
 
+import java.util.Set;
+
 @AllArgsConstructor
 public class FlashcardFacade {
-    private FlashcardRepository flashcardRepository;
+    private FlashcardService flashcardService;
+    private QuestionTagService questionTagService;
     private ModelMapper modelMapper;
 
     public FlashcardSaveResponseDto saveFlashcard(FlashcardDto flashcardDto) {
         Flashcard flashcard = modelMapper.map(flashcardDto, Flashcard.class);
         try{
-            Flashcard savedFlashcard = flashcardRepository.save(flashcard);
+            Flashcard savedFlashcard = flashcardService.saveFlashcard(flashcard);
+            Set<QuestionTag> questionTagsSet = questionTagService.getQuestionTagsSet(flashcardDto.getTagsList(), flashcard);
+            savedFlashcard.setQuestionTagsList(questionTagsSet);
+            flashcardService.saveFlashcard(savedFlashcard);
```

Comment:
saveFlashcard is enough? We don't need to save QuestionTags?

Reply:
It works because on questionTagsSet in Flashcard is set cascadeType=ALL (but maybe it won't work after deleting flashcard) -> taken under consideration